src/pybind/mgr/dashboard/frontend/src/app/shared/services/favicon.service.ts

```diff
@@ -0,0 +1,67 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable, OnDestroy } from '@angular/core';
+
+import { Subscription } from 'rxjs';
+
+import { Color } from '../enum/color.enum';
+import { SummaryService } from './summary.service';
+
+@Injectable()
+export class FaviconService implements OnDestroy {
+  sub: Subscription;
+  oldStatus: string;
+  url: string;
+
+  constructor(
+    @Inject(DOCUMENT) private document: HTMLDocument,
+    private summaryService: SummaryService
+  ) {}
+
+  init() {
+    this.url = this.document.getElementById('cdFavicon')?.getAttribute('href');
+
+    this.sub = this.summaryService.subscribe((summary) => {
+      this.changeIcon(summary.health_status);
+    });
+  }
+
+  changeIcon(status?: string) {
+    if (status === this.oldStatus) {
+      return;
+    }
+
+    this.oldStatus = status;
+
+    const favicon = this.document.getElementById('cdFavicon');
+    const faviconSize = 16;
+
+    const canvas = this.document.createElement('canvas');
+    canvas.width = faviconSize;
+    canvas.height = faviconSize;
+
+    const context = canvas.getContext('2d');
+    const img = this.document.createElement('img');
+    img.src = this.url;
+
+    img.onload = () => {
+      // Draw Original Favicon as Background
+      context.drawImage(img, 0, 0, faviconSize, faviconSize);
+
+      // Draw Notification Circle
```

Comment:
I would like to suggest to cut-out the circle area with radius + 1 for a better look.

Before:
![screenshot](https://user-images.githubusercontent.com/1897962/87654394-1f557d80-c757-11ea-9747-d383e1561b31.png)

After:
![screenshot2](https://user-images.githubusercontent.com/1897962/87654407-241a3180-c757-11ea-8070-bbbef4ecfc51.png)

![screenshot3](https://user-images.githubusercontent.com/1897962/87654417-28464f00-c757-11ea-9531-c7b35af5ae3e.png)

```suggestion
      // Cut notification circle area
      context.save();
      context.globalCompositeOperation = 'destination-out';
      context.beginPath();
      context.arc(canvas.width - faviconSize / 4, faviconSize / 4, (faviconSize / 4) + 1, 0, 2 * Math.PI);
      context.fill();
      context.restore();

      // Draw notification circle
```

Reply:
Done. Went with 1px.